src/types.tsx

```diff
@@ -610,8 +609,13 @@ export interface ScreenContainerProps extends ViewProps {
 }
 
 export interface GestureDetectorBridge {
+  /**
+   * Callback to attach into ScreenStack's useEffect() from ScreenGestureDetector that wraps the stack.
+   *
+   * @param stackRef holds a reference to an instance of ScreenStackNativeComponent
+   */
   stackUseEffectCallback: (
-    stackRef: React.MutableRefObject<React.Ref<NativeStackNavigatorProps>>,
+    stackRef: React.MutableRefObject<React.Ref<View>>,
```

Comment:
This is potentially a breaking change. Given that we never really downstreamed the feature it is used for I give a green light and take the responsibility here.